### PR TITLE
[ROCm] Fix AITER AR+RMSNorm no-residual fusion

### PR DESCRIPTION
**Pull request by akii96**
## Purpose

Fix the ROCm AITER allreduce + RMSNorm fusion for the no-residual pattern.

`AiterAllreduceFusedRMSNormPattern` replaces an allreduce followed by RMSNorm without a residual input. However, the AITER fused kernel computes RMSNorm over `allreduce(input) + residual`, so the synthetic residual for this pattern must be zero.

The AITER replacement used `torch.empty_like(input)`, which can add uninitialized memory into the layer output. This PR changes it to `torch.zeros_like(input)`, matching the existing FlashInfer no-residual fusion patterns in the same file.

This restores MiniMax-M2.5 GSM8K accuracy while keeping the AITER fusion enabled.

## Test Plan

Serve MiniMax-M2.5 with ROCm AITER and allreduce RMSNorm fusion enabled:

```bash
vllm serve MiniMaxAI/MiniMax-M2.5 \
  --tensor-parallel-size 4 \
  --attention-backend ROCM_AITER_UNIFIED_ATTN \
  --max-model-len 12288 \
  --block-size 64 \
  --max-num-seqs 512 \
  --max-num-batched-tokens 32768 \
  --gpu-memory-utilization 0.95 \
  --performance-mode balanced \
  --async-scheduling \
  --no-enable-prefix-caching \
  --kv-cache-dtype auto \
  --compilation-config '{"mode":3}'
```
  
  ## Test Result
  
  Before this fix, GSM8K accuracy collapsed with ROCm AITER allreduce RMSNorm fusion enabled 
  
  |Tasks|Version|     Filter     |n-shot|  Metric   |   |Value|   |Stderr|
|-----|------:|----------------|-----:|-----------|---|----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |    0|±  |     0|
|     |       |strict-match    |     5|exact_match|↑  |    0|±  |     0|
  
  
  After this fix:
  
  
  
  |Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.9515|±  |0.0059|
|     |       |strict-match    |     5|exact_match|↑  |0.9454|±  |0.0063|

### CODE REVIEW

**Review by dllehr-amd — APPROVED** [review]
Looks good!  Nice catch @akii96 !

**Review by gshtras — APPROVED** [review]

**Comment by rbrugaro-amd** [comment]
@gshtras @dllehr-amd  we created https://github.com/vllm-project/vllm/pull/41767 few days back addressing this issue. The empty/zeros fix on this PR addressed the accuracy but without the `variance_size_override` the fusion was not getting picked up correctly. Can you trigger the CICD test on 41767?

**Review by tjtanaa — APPROVED** [review]
LGTM

**Review by ProExpertProg — APPROVED** [review]

**Review by mgoin — APPROVED** [review]
